blog/views.py

```diff
@@ -15,13 +15,34 @@ def homepage(request):
 
 def post_details(request, slug):
     post = Post.objects.filter(slug=slug)
+    p = get_object_or_404(Post, slug=slug)
 
     if post.exists():
         post = post[0]
     else:
         return HttpResponse("Page not found")
 
-    context = {"post": post}
+    comments = post.comments.filter(approve=True)
+    comment_author = request.user
+    new_comment = None
+
+    if request.method == "POST":
+        comment_form = AddCommentForm(data=request.POST)
+        if comment_form.is_valid():
+            new_comment = comment_form.save(commit=False)
+            new_comment.post = p
+            new_comment.user = comment_author
+            new_comment.save()
+            comment_form = AddCommentForm()
```

Comment:
This function allocates crucial comment information. In which post comment is located and by which user.